decoder.go

```diff
@@ -240,20 +311,82 @@ transform:
 
 		switch d.format {
 		case FormatYenc:
-			d.processYenc(line)
-			goto transform
+			// Header/trailer lines
+			if bytes.HasPrefix(line, []byte("=ybegin ")) ||
+				bytes.HasPrefix(line, []byte("=ypart ")) ||
+				bytes.HasPrefix(line, []byte("=yend ")) {
+				d.processYenc(line)
+				goto transform
+			}
+			// If we're in the body, decode this line
+			if d.body {
+				// Remove trailing \r\n for decoding
+				bodyLine := line
+				if len(bodyLine) >= 2 && bodyLine[len(bodyLine)-2] == '\r' && bodyLine[len(bodyLine)-1] == '\n' {
+					bodyLine = bodyLine[:len(bodyLine)-2]
+				}
+
+				dlog(d.debugSpam, "DecodeIncremental input: %q\n", bodyLine)
+				nd, ns, end, derr := DecodeIncremental(dst[nDst:], bodyLine, &d.State)
+				if derr != nil && derr != io.EOF {
+					dlog(always, "ERROR in rapidyenc.DecodeIncremental: nd=%d ns=%d end=%v err=%v\n", nd, ns, end, derr)
+				}
```

Comment:
This is decoding line-by-line, which is slow.

Rapidyenc was made to handle raw NNTP content with `\r\n`, there is a penalty using CGO so we want to try to minimise calls and process large chunks of data - I didn't see any benefit beyond 4KiB.
You lose the ability to log the line content but I'm not sure how useful that is anyway as it's not really human readable.